src/views/checkups/network/components/CheckupsNetworkStatusIcon.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC } from 'react';
+
+import { IoK8sApiBatchV1Job, IoK8sApiCoreV1ConfigMap } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { NO_DATA_DASH } from '@kubevirt-utils/resources/vm/utils/constants';
+import { CheckCircleIcon, ExclamationCircleIcon, SyncAltIcon } from '@patternfly/react-icons';
+
+import { STATUS_SUCCEEDED } from '../utils/utils';
+
+import './checkups-network-status.icon.scss';
+
+type CheckupsNetworkStatusIconProps = {
+  configMap?: IoK8sApiCoreV1ConfigMap;
+  job: IoK8sApiBatchV1Job;
+  onlyJob?: boolean;
+};
+const CheckupsNetworkStatusIcon: FC<CheckupsNetworkStatusIconProps> = ({
+  configMap,
+  job,
+  onlyJob = false,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const statusJob =
+    (job?.status?.succeeded === 1 && 'done') ||
+    (job?.status?.active === 1 && 'running') ||
+    (job?.status?.succeeded === 0 && 'failed');
+
+  const statusConfigMap =
+    (configMap?.data?.[STATUS_SUCCEEDED] === 'true' && 'done') ||
+    ((configMap?.data?.[STATUS_SUCCEEDED] === 'false' || statusJob === 'failed') && 'failed') ||
+    (configMap?.data?.[STATUS_SUCCEEDED] === undefined && statusJob === 'done' && 'failed') ||
+    (configMap?.data?.[STATUS_SUCCEEDED] === undefined && job?.status?.active === 1 && 'running');
```

Comment:
Here we can have two functions  maybe?

Reply:
hmm i will think about something because its ugly lol

Reply:
yes, i will think about something , a switch , if else (which I don't like - it not pretty and confusion IMHO), function or something else.. :)